draft-ietf-quic-qpack.md

```diff
@@ -270,8 +279,8 @@ immediately. A stream becomes unblocked when the greatest absolute index in the
 dynamic table becomes greater than or equal to the Largest Reference for all
 header blocks the decoder has started reading from the stream.  If a decoder
 encounters a header block where the actual largest reference is not equal to the
-Largest Reference declared in the prefix, it MAY treat this as a stream error of
-type HTTP_QPACK_DECOMPRESSION_FAILED.
+Largest Reference declared in the prefix, it MUST treat this as an error
```

Comment:
My reason for changing this to MUST is the following: if this is a MAY, potentially implementation get deployed that do not signal an error, which allows implementation that have a larger Largest Reference declared to be deployed, and once there are enough implementations out there, if a new one signals an error in conformance to the specification, it will break things, and we'll be stuck with an ecosystem that diverged from the spec.

An encoder should never declare a Largest Reference larger than the actual largest reference, let's encourage decoders to enforce it.

(Note that smaller Largest Reference values cannot be sent because there is already a MUST clause for that in 2.2.2.)

Reply:
See the discussion of MUST vs. MAY for this case in https://github.com/quicwg/base-drafts/issues/1404.

I was for MUST here previously.  I think the argument against it was that it's kind of annoying to check it potentially, and some organizations require all MUSTs to be implemented or have a 'really good reason' (TM).

Reply:
I understand.  Sorry, I was not aware of that previous discussion.  Let me revert this to MAY.